modules/axtask/src/task.rs

```diff
@@ -292,6 +298,29 @@ impl TaskInner {
     }
 }
 
+impl TaskInner {
+    /// Allocate a key
+    pub fn alloc(&self, destr_function: Option<DestrFunction>) -> Option<usize> {
+        unsafe { KEYS.lock().alloc(destr_function) }
+    }
+    /// Get the destructor function of a key
+    pub fn free(&self, key: usize) {
```

Comment:
`free_key` or `delete_key`. And `pthread_key_delete` function should set error number if key is out of range.
> ERRORS
> The pthread_key_delete() function may fail if:
> 
> [EINVAL]
> The key value is invalid.